### PR TITLE
Problem: REST API docs are poorly organized

### PR DESCRIPTION
**Pull request by dkliban**
Solution: Break up REST API docs into sections

This patch modifies the OpenAPI schema. It introduces the concept of Tags. Each tag represents a type
of resource in Pulp. E.g. Artifacts, Content, Remotes, Publishers, etc.

This patch also improves the 'summary' field for each operation also. This is achieved with a new
custom inspector class that drf_yasg can use to build the OpenAPI schema.

This patch also adds a new 'restapi.html' page to the static docs. This new page uses ReDoc and the
OpenAPI schema to display the REST API docs.

Required PR: pulp/pulp_file#202

re: #4104
https://pulp.plan.io/issues/4104

### CODE REVIEW

File: pulpcore/app/urls.py [hunk 1]
```diff
@@ -123,14 +123,14 @@ def __repr__(self):
     url(r'^auth/', include('rest_framework.urls')),
 ]
 
+api_info = openapi.Info(
+   title="Snippets API",
```
Comment (asmacdo) [inline pulpcore/app/urls.py]: copy/paste error

File: pulpcore/app/settings.py [hunk 2]
```diff
@@ -216,3 +216,9 @@
 CONTENT_PATH_PREFIX = settings.get('CONTENT_PATH_PREFIX', '/pulp/content/')
 
 PROFILE_STAGES_API = settings.get('PROFILE_STAGES_API', False)
+
+SWAGGER_SETTINGS = {
+    'DEFAULT_GENERATOR_CLASS': 'pulpcore.app.openapigenerator.PulpOpenAPISchemaGenerator',
+    'DEFAULT_AUTO_SCHEMA_CLASS': 'pulpcore.app.openapigenerator.PulpAutoSchema',
+    'DEFAULT_INFO': 'pulpcore.app.urls.api_info',
+}
```
Comment (bmbouter) [inline pulpcore/app/settings.py]: newline at the end here plz

**Comment by bmbouter** [comment]
I tested this just now and this PR really cleaned up the live docs page for me! Thanks @dkliban :+1: 

Here is a screenshot of the nice collapset and named interface:

Screenshot from 2019-04-04 10-29-52

File: pulpcore/app/urls.py (hunk 1 above)
Comment (dkliban) [reply pulpcore/app/urls.py]: fixed it. thank you!

File: pulpcore/app/settings.py (hunk 2 above)
Comment (dkliban) [reply pulpcore/app/settings.py]: added new line

File: docs/Makefile [hunk 3]
```diff
@@ -50,7 +50,7 @@ diagrams:
 	mv diagrams_src/*.png $(DIAGRAM_BUILD_DIR)/
 
 html:
-	curl -o api.yaml "http://localhost:8000/pulp/api/v3/docs/api.yaml"
```
Comment (asmacdo) [inline docs/Makefile]: We need to add `mkdir -p _build/html/` before the next line.

Comment (asmacdo) [reply docs/Makefile]: Actually, `mkdir -p ${BUILDDIR}/html/`

**Comment by asmacdo** [comment]
Initially had some trouble seeing this locally. `make html` and browsing files failed for me, but the extra step of forwarding a port and running a python server from the docs dir made these docs viewable. 

I also tested that the docs page works correctly even if pulp-api is shut down, provided that it was on during the docs build.

The Travis error is the same as the error I had. `Failed to create the file _build/html/api.json: No such file or directory` which I solved with the suggested line addition in the Makefile.

File: pulpcore/app/openapigenerator.py [hunk 4]
```diff
@@ -160,3 +187,114 @@ def get_parameter_name(model):
...
+        summary, description = self.get_summary_and_description()
+        security = self.get_security()
+        assert security is None or isinstance(security, list), "security must be a list of security requirement objects"
+        deprecated = self.is_deprecated()
+        tags = self.get_tags(operation_keys)
+
+        responses = self.get_responses()
+        summary = self.get_summary(operation_keys)
+        #import pydevd
+        #pydevd.settrace('localhost', port=3013, stdoutToServer=True, stderrToServer=True)
```
Comment (asmacdo) [inline pulpcore/app/openapigenerator.py]: dont forget to remove this

File: docs/Makefile [hunk 5]
```diff
@@ -50,7 +50,8 @@ diagrams:
...
+	mkdir -p _build/html
+	curl -o _build/html/api.json "http://localhost:8000/pulp/api/v3/docs/?format=openapi"
```
Comment (asmacdo) [inline docs/Makefile]: better to use https://github.com/pulp/pulpcore/pull/67/files#diff-68a7f9edd4ffbca6adb99e2c165cfd52L8 both places

File: pulpcore/app/openapigenerator.py (hunk 4 above)
Comment (dkliban) [reply pulpcore/app/openapigenerator.py]: i did.

File: docs/Makefile (hunk 5 above)
Comment (dkliban) [reply docs/Makefile]: I substituted for the variable. Thanks!

**Pull request merged**